src/polyswarm_api/engine_resolver.py

```diff
@@ -0,0 +1,39 @@
+import aiohttp
+import asyncio
+import logging
+
+
+class EngineResolver(object):
+
+    def __init__(self, api_addr):
+        self.logger = None
+        self.api_addr = api_addr
+        self.engine_map = None
+        self.run()
+
+    @classmethod
+    def get_logger_name(cls):
+        return cls.__name__
+
+    def get_engine_name(self, eth_pub):
+        return self.engine_map.get(eth_pub.lower(), eth_pub) if self.engine_map is not None else ''
+
+    async def get_engines(self):
+        try:
+            self.logger.debug("Begin engine name polling")
+            headers = {'content-type': 'application/json'}
+            async with aiohttp.ClientSession(headers=headers) as session:
+                async with session.get('{}/{}'.format(self.api_addr, 'v1/microengines/list')) as response:
+                    result = await response.json()
```

Comment:
Don't go straight to `json()`, need to check endpoint response code to see if request failed or not.

Currently, this also breaks because it's 404'ing and returning a text/plain mimetype. So need endpoints up before we merge this in.

Reply:
yeah I forgot that it wasn't up in prod yet. WIll make those other changes for you too.

Reply:
Ah, v1 shouldn't be in the path. V1 is already included in the addr.

Reply:
I was testing in stage as well. but yeah,  no v1 and also use the `f"{my_var}"` formatting that we're standardizing on now instead of `.format()`